test/hardware-fakes/src/fake_memory.c

```diff
@@ -23,12 +23,41 @@ static uint8_t _memory_shared_data[FLASH_SHARED_DATA_LEN] = {0};
 static uint8_t _memory_app_data[FLASH_APPDATA_LEN] = {0};
 static uint8_t _memory_smarteeprom[SMARTEEPROM_RESERVED_FLASH_PAGES * FLASH_PAGE_SIZE] = {0};
 
+static void _init_file_if_needed(
+    const char* base_path,
+    const char* suffix,
+    const uint8_t* data,
+    size_t data_size)
+{
+    char path[512];
+    snprintf(path, sizeof(path), "%s_%s", base_path, suffix);
```

Comment:
I think you should check the return value of `snprintf`. If it is 512 you should exit/abort and report an error to the user. Better than writing in the wrong place on the disk.